collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2830,8 +2836,18 @@ protected void onStop() {
         super.onStop();
     }
 
-    private void sendSavedBroadcast() {
-        sendBroadcast(new Intent("org.odk.collect.android.FormSaved"));
+    void enqueueFinalizedForm() {
```

Comment:
It's also important that if the setting is set to Wifi only, the forms only send when wifi becomes available and the same for data.

Reply:
Right now every single form finalization results in the auto send job being enqueued, correct? And then if there's connectivity, the job immediately checks if `auto-send` is enabled and completes if not. But if there's no connectivity, the job stays scheduled until connectivity becomes available and then if the check sees that `auto-send` is not enabled, it completes. Is that correct? It doesn't seem right that a job would stay enqueued if `auto-send` is off in the first place.

Reply:
I agree. Thanks for spotting this.

Reply:
Kind of related -- where is it enforced that only one of these jobs is ever enqueued? As you said, since the job is to send all forms, I believe there should only ever be either 0 or 1 enqueued. So if a form is finalized while connectivity is off and auto-send is on, that should enqueue the work. Then if another form is finalized while connectivity is still off and auto-send is still on, nothing should happen because the work is already enqueued.

Reply:
Yes, that's how the `WorkManager`s api is currently being utilized. Referenced that call in another comment.